client/my-sites/plans-grid/components/plans-2023-tooltip.tsx

```diff
@@ -1,11 +1,17 @@
 import styled from '@emotion/styled';
 import { TranslateResult } from 'i18n-calypso';
-import { Dispatch, PropsWithChildren, SetStateAction, useEffect, useRef, useState } from 'react';
+import { Dispatch, PropsWithChildren, SetStateAction, useRef } from 'react';
 import Tooltip from 'calypso/components/tooltip';
 import { hasTouch } from '../lib/touch-detect';
 
-const HoverAreaContainer = styled.span`
-	max-width: 220px;
+// TODO:
+// The hover area container should ideally fit the dimensions of the child components
+// However, it's not easy to find a configuration that fits all possible display,
+// e.g. the feature copies in the plans table is an `inline` element, the downgrade button
+// is a `inline-block` element having 100% width. Looking into @wordpress/tooltip, I think
+// there is a way, but this intermediate solution seems to provide a less intrusive solution for the current needs.
+const HoverAreaContainer = styled.div< { display?: 'block' | 'inline-block' } >`
+	display: ${ ( { display = 'block' } ) => display };
```

Comment:
To further expand on the inline explanation here, this PR replaces the `DummyDisabledButton` by reusing the same buttone component. The current tooltip didn't play well for that:
<img width="968" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1842898/654d1ecb-b5bb-4261-94b7-f7db6932abe0">


`HoverAreaContainer` used to be a `span`, which also didn't seem right since a span cannot be a container. The max-width is also not needed since it should be as wide as the contained components, and `max-width` doesn't work with an `inline` element anyway. Thus I've changed it into a div and restrict the display mode to `inline-block` and `block` only.